pkg/core/function.go

```diff
@@ -328,6 +353,8 @@ func (c *client) BuildFunction(options BuildFunctionOptions, log io.Writer) erro
 		return err
 	}
 
+	gen := s.Spec.Generation
```

Comment:
It's a k8s API field - see the doc Eric linked to.

"Some resources report the observedGeneration, which is the generation most recently observed by the component responsible for acting upon changes to the desired state of the resource. This can be used, for instance, to ensure that the reported status reflects the most recent desired status."

Reply:
I think @glyn is asking about why this is part of `Spec`.